ompi/request/req_wait.c

```diff
@@ -361,11 +375,14 @@ int ompi_request_default_wait_some(size_t count,
     ompi_request_t **rptr = NULL;
     ompi_request_t *request = NULL;
     ompi_wait_sync_t sync;
-
+    size_t sync_sets = 0, sync_unsets = 0;
```

Comment:
As suggested in my previous comment you can infer the sync_sets and sync_unsets from the other variables without having to update them continuously.